packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -530,8 +530,11 @@ - (void)disposeSansEventChannel {
   [currentItem removeObserver:self forKeyPath:@"playbackLikelyToKeepUp"];
   [currentItem removeObserver:self forKeyPath:@"playbackBufferEmpty"];
   [currentItem removeObserver:self forKeyPath:@"playbackBufferFull"];
+  [self.player removeObserver:self forKeyPath:@"rate"];
 
   [self.player replaceCurrentItemWithPlayerItem:nil];
+  _player = nil;
```

Comment:
why is this required?

Reply:
When applied the original fix, the application using the video_player started crashing while doing the Hot Restart with Flutter attached in debug mode. The reason for this crash is that the player is no longer KVO subscribed to property rate, therefore remove observer calls crashes.

With additional logging I was able to confirm that disposeSansEvent method is called twice on the same player. The issue is not present in the original code, because of replace current item with nil call that happens in the dispose (the line before I nullify the player). For the subsequent call, the player item is nullified, therefore remove observer call is no-op. I applied the same style of the solution to the player.

Reply:
when it's called twice, should we short circuit up front? 

```
- (void)dispose... {
  if (!_player) {
    return;
  } 
  ... 
}
```

Reply:
I'll short circuit it with:

```objc
- (void)disposeSansEventChannel {
  if (_disposed) {
    return;
  }
  _disposed = YES;
// ...
}
```

as `_disposed` is already set there.

Reply:
makes sense. Can you also add some comments here?

Reply:
@hellohuanlin I've updated the code according to our discussion. I've added a comment on the short circuit method.

I don't think removing the observer requires additional comments as the API specifies to clear any observations you've subscribed to in the first place. LMK if you'd add anything else, please 🙏